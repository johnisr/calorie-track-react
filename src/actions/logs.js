import database from '../firebase/firebase';
import moment from 'moment';

export const defaultLog = {
  date: 0,
  weight: 0,
  unit: 'lb',
  foods: [],
};

export const addLog = (log) => ({
  type: 'ADD_LOG',
  log,
});

export const startAddLog = (data = {}) => {
  return async (dispatch, getState) => {
    const { uid } = getState().auth;
    const {
      weight = 0,
      unit = 'lb',
      foods = [],
      date = moment(),
    } = data;
    const log = { weight, unit, foods, date };

    try {
      await database.ref(`users/${uid}/logs/${date}`).set(log);
      dispatch(addLog(log));
    } catch (e) {

    };
  };
};

export const removeLog = (date) => ({
  type: 'REMOVE_LOG',
  date,
});

export const startRemoveLog = (date) => {
  return async (dispatch, getState) => {
    const { uid } = getState().auth;
    try {
      await database.ref(`users/${uid}/logs/${date}`).remove();
      dispatch(removeLog(date));
    } catch (e) {

    };
  };
};

export const editLog = (date, updates) => ({
  type: 'EDIT_LOG',
  date,
  updates,
});

export const startEditLog = (date, updates) => {
  return async (dispatch, getState) => {
    const { uid } = getState().auth;
    try {
      await database.ref(`users/${uid}/logs/${date}`).update(updates);
      dispatch(editLog(date, updates));
    } catch (e) {
      
    }
  };
};

export const setLogs = (logs) => ({
  type: 'SET_LOGS',
  logs,
});

export const startSetLogs = () => {
  return async (dispatch, getState) => {
    const { uid } = getState().auth;

    try {
      const snapshot = await database.ref(`users/${uid}/logs`).once('value');
      const logs = [];
      snapshot.forEach((log) => {
        logs.push(log.val());
      });
      dispatch(setLogs(logs));
    } catch (e) {

    }
  };
};