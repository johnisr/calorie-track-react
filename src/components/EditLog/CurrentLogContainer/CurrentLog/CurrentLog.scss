@import '../../../../styles/base';

.currentLog {
  display: flex;
  justify-content: space-between;
  margin-top: 2rem;

  &__btn {
    display: flex;
    align-items: center;
    
    & > * {
      margin-left: 2rem;
    }
  }

  &__weight {
    display: flex;
  }

  &__radio-group {
    margin-left: 2rem;
  }
  
  &__radio-input {
    display: none;
  }

  &__radio-label {
    cursor: pointer;
    position: relative;
    padding-left: 3.5rem;
  }

  &__radio-button {
    height: 3rem;
    width: 3rem;
    border: .5rem solid $color-grey-dark-3;
    border-radius: 50%;
    display: inline-block;
    position: absolute;
    left: 0;
    top: -.5rem;

    &::after {
      content: "";
      display: block;
      height: 1.3rem;
      width: 1.3rem;
      border-radius: 50%;
      
      position: absolute;
      top: 50%;
      left: 50%;
      transform: translate(-50%, -50%);
      
      background-color: $color-grey-dark-3;
      opacity: 0;
      transition: opacity .2s;
    }
  }

  &__radio-input:checked ~ &__radio-label &__radio-button::after {
    opacity: 1;
  }

  &__input {
    font-size: $font-size-body;
    width: 100%;
    
    padding: .5rem 1rem;

    border: none;
    border-bottom: 3px solid transparent;
    display: block;

    &:focus {
      outline: none;
      box-shadow: 0 1rem 2rem rgba(0,0,0, .1);
      border-bottom: 3px solid $color-grey-dark-3;
    }

    &--invalid {
      border-bottom: 3px solid red;
    }
    
  }

  &__label {
    display: block;
    transition: all .3s;
    margin-left: 1rem;
    transform: translateY(-6rem);
  }

  &__input:placeholder-shown + &__label {
    opacity: 0;
    visibility: hidden;
    transform: translateY(-3.5rem);
  }

  &__input:focus + &__label {
    opacity: 1;
    visibility: visible;
    transform: translateY(-6rem);
  }
}