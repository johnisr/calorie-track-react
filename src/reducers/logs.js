export default (state = [], action) => {
  switch (action.type) {
    case 'ADD_LOG': {
      return [...state, action.log];
    }
    case 'REMOVE_LOG': {
      return state.filter(({ date }) => !(date === action.date));
    }
    case 'EDIT_LOG': {
      const newState = state.map((log) => {
        if (log.date === action.date) {
          return {
            ...log,
            ...action.updates,
          };
        }
        return log;
      });
      return newState;
    }
    case 'SET_LOGS': {
      return action.logs;
    }
    default: {
      return state;
    }
  }
};