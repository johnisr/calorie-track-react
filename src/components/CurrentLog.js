import React from 'react';
import { connect } from 'react-redux';
import { removeFoodFromCurrentLog } from '../actions/currentLog';
import FoodListDisplayCurrentLog from './FoodListDisplayCurrentLog';

class CurrentLog extends React.Component {
  handleRemove = (food) => {
    this.props.removeFoodFromCurrentLog(food.index);
  };
  render() {
    return (
      <div>
        <FoodListDisplayCurrentLog />
      </div>
    );
  }
};

const mapStateToProps = (state) => ({
  currentLog: state.currentLog,
});

const mapDispatchToProps = (dispatch) => ({
  removeFoodFromCurrentLog: (index) =>  dispatch(removeFoodFromCurrentLog(index)),
});

export default connect(mapStateToProps, mapDispatchToProps)(CurrentLog);