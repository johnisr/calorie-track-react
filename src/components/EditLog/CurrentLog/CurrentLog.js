import React from 'react';
import { connect } from 'react-redux';
import { editCurrentEditLog, removeCurrentEditLog } from '../../../actions/currentLog';
import FoodListDisplayCurrentLogContainer from './FoodListDisplayCurrentLog/FoodListDisplayCurrentLogContainer';
import { startEditLog, startRemoveLog } from '../../../actions/logs';
import { startEditFood } from '../../../actions/foods';

class CurrentLog extends React.Component {
  updateTimesUsed() {
    const foodUpdates = {};
    const oldLog = this.props.logs.filter(log => log.date === this.props.currentLog.date)[0];
    
    // Check if oldLog.foods exist because firebase doesn't store empty arrays
    if (oldLog.foods) {
      oldLog.foods.forEach((food) => {
        if (foodUpdates.hasOwnProperty(food.id)) {
          foodUpdates[food.id] = foodUpdates[food.id] - 1;
        } else {
          foodUpdates[food.id] = -1;
        }
      });
    };
    
    if (this.props.currentLog.foods) {
      this.props.currentLog.foods.forEach((food) => {
        if (foodUpdates.hasOwnProperty(food.id)) {
          foodUpdates[food.id] = foodUpdates[food.id] + 1;
        } else {
          foodUpdates[food.id] = 1;
        }
      });
    }

    if (this.props.foods) {
      this.props.foods.forEach((food) => {
        if (foodUpdates[food.id]) {
          this.props.startEditFood(
            food.id, 
            { timesUsed: food.timesUsed + foodUpdates[food.id] } 
          );
        }
      });
    }
  }

  isNumberWithAtMostTwoDecimals = (str) => {
    return str.match(/^\d{1,}?(\.\d{0,2})?$/);
  }
  handleWeightChange = (e) => {
    const weight = e.target.value;
    if (!weight || this.isNumberWithAtMostTwoDecimals(weight)) {
      this.props.editCurrentEditLog({ weight });
    }
  }
  handleLbsSelected = () => {
    const unit = 'lbs';
    this.props.editCurrentEditLog({ unit });
  }
  handleKgsSelected = () => {
    const unit = 'kgs';
    this.props.editCurrentEditLog({ unit });
  }
  handleSubmit = async () => {
    const weight = this.props.currentLog.weight !== '' ? parseFloat(this.props.currentLog.weight, 10) : 0;

    const total = this.props.currentLog.foods.reduce((prev, curr) => ({
      protein: prev.protein + curr.protein,
      carbohydrates: prev.carbohydrates + curr.carbohydrates,
      fat: prev.fat + curr.fat,
      calories: prev.calories + curr.calories,
    }), {protein: 0, carbohydrates: 0, fat: 0, calories: 0});
    await this.props.editCurrentEditLog({ total });

    this.props.startEditLog(this.props.currentLog.date, {
      date: this.props.currentLog.date,
      unit: this.props.currentLog.unit,
      foods: this.props.currentLog.foods,
      weight,
      total: {
        carbohydrates: +(total.carbohydrates).toFixed(2),
        protein: +(total.protein).toFixed(2),
        fat: +(total.fat).toFixed(2),
        calories: +(total.calories).toFixed(2),
      }
    });
    this.updateTimesUsed();
    
    this.props.removeCurrentEditLog();
    this.props.history.push('/LogDashboard');
  }
  handleRemove = () => {
    this.props.startRemoveLog(this.props.currentLog.date);
    this.props.removeCurrentEditLog();
    this.props.history.push('/LogDashboard');
  }
  handleExit = () => {
    this.props.removeCurrentEditLog();
    this.props.history.push('/LogDashboard');
  }
  render() {
    return (
      <div className="container">
        <div>
          <FoodListDisplayCurrentLogContainer />
        </div>
        <section className="section-center-6-start-end">
          <div className="currentLog">
            <div className="currentLog__weight">
              <div>
                <input 
                  name="weight"
                  className="form__input"
                  type="text"
                  placeholder="Weight"
                  value={this.props.currentLog.weight}
                  onChange={this.handleWeightChange} 
                />
                <label className="form__label" htmlFor="weight">Weight</label>
              </div>
              <div className="currentLog__radio-group">
                <input
                  className="currentLog__radio-input"
                  type="radio"
                  name="unit"
                  id="lb"
                  value="lbs"
                  onChange={this.handleLbsSelected}
                  checked={this.props.currentLog.unit === "lbs"}
                />
                <label htmlFor="lb" className="currentLog__radio-label">
                  <span className="currentLog__radio-button"></span>
                  lbs
                </label>
              </div>
              <div className="currentLog__radio-group">
              <input 
                type="radio"
                className="currentLog__radio-input"
                name="unit"
                value="kgs"
                id="kg"
                onChange={this.handleKgsSelected}
                checked={this.props.currentLog.unit === "kgs"}
              />
                <label htmlFor="kg" className="currentLog__radio-label">
                  <span className="currentLog__radio-button"></span>
                  kgs
                </label>
              </div>
            </div>
            <div className="currentLog__btn">
              <div>
                <button className="btn btn--form btn--negative" onClick={this.handleRemove}>Remove</button>
              </div>
              <div>
                <button className="btn btn--form btn--positive" onClick={this.handleSubmit}>Submit</button>
              </div>
            </div>
          </div>
          </section>
      </div>
    );
  }
};

const mapStateToProps = (state) => ({
  currentLog: state.currentLog,
  logs: state.logs,
  foods: state.foods,
});

const mapDispatchToProps = (dispatch) => ({
  editCurrentEditLog: (updates) => dispatch(editCurrentEditLog(updates)),
  removeCurrentEditLog: () => dispatch(removeCurrentEditLog()),
  startEditLog: (date, log) => dispatch(startEditLog(date, log)),
  startEditFood: (id, addition) => dispatch(startEditFood(id, addition)),
  startRemoveLog: (date) => dispatch(startRemoveLog(date)),
});

export default connect(mapStateToProps, mapDispatchToProps)(CurrentLog);