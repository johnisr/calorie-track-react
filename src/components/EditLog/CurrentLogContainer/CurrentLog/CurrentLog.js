import React from 'react';
import './CurrentLog.css';

const currentLog = (props) => (
  <div className="currentLog">
    <div className="currentLog__weight">
      <div>
        <input 
          name="weight"
          className="currentLog__input"
          type="text"
          placeholder="Weight"
          value={props.currentLog.weight}
          onChange={props.onWeightChange} 
        />
        <label className="currentLog__label" htmlFor="weight">Weight</label>
      </div>
      <div className="currentLog__radio-group">
        <input
          className="currentLog__radio-input"
          type="radio"
          name="unit"
          id="lb"
          value="lbs"
          onChange={props.onLbsSelected}
          checked={props.currentLog.unit === "lbs"}
        />
        <label htmlFor="lb" className="currentLog__radio-label">
          <span className="currentLog__radio-button"></span>
          lbs
        </label>
      </div>
      <div className="currentLog__radio-group">
      <input 
        type="radio"
        className="currentLog__radio-input"
        name="unit"
        value="kgs"
        id="kg"
        onChange={props.onKgsSelected}
        checked={props.currentLog.unit === "kgs"}
      />
        <label htmlFor="kg" className="currentLog__radio-label">
          <span className="currentLog__radio-button"></span>
          kgs
        </label>
      </div>
    </div>
    <div className="currentLog__btn">
      <div>
        <button type="button" className="btn btn--form btn--negative" onClick={props.onRemove}>Remove</button>
      </div>
      <div>
        <button className="btn btn--form btn--positive" onClick={props.onSubmit}>Submit</button>
      </div>
    </div>
  </div>
);

export default currentLog;