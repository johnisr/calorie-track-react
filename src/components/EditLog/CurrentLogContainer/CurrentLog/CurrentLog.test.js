import React from 'react';
import { shallow } from 'enzyme';
import CurrentLog from './CurrentLog';
import logs from '../../../../tests/fixtures/logs';

it('should render currentLog', () => {
  const wrapper = shallow(<CurrentLog
    currentLog={logs[0]}
    onWeightChange={jest.fn()}
    onLbsSelected={jest.fn()}
    onKgsSelected={jest.fn()}
    onRemove={jest.fn()}
    onSubmit={jest.fn()}
  />);
  expect(wrapper).toMatchSnapshot();
});
