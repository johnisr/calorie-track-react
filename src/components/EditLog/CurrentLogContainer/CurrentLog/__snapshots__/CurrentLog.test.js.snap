// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render currentLog 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <currentLog
    currentLog={
        Object {
            "date": -154800000,
            "foods": Array [
              Object {
                "amount": 24,
                "calories": 163,
                "carbohydrates": 2.5,
                "createdAt": -1000,
                "fat": 14,
                "id": "3",
                "name": "Almonds",
                "protein": 6,
                "timesUsed": 2,
                "unit": "pieces",
              },
              Object {
                "amount": 200,
                "calories": 205,
                "carbohydrates": 44,
                "createdAt": 0,
                "fat": 0.4,
                "id": "1",
                "name": "Rice (Jasmine)",
                "protein": 4.2,
                "timesUsed": 0,
                "unit": "grams",
              },
              Object {
                "amount": 100,
                "base": Object {
                  "amount": 100,
                  "calories": 165,
                  "carbohydrates": 0,
                  "fat": 3.6,
                  "protein": 31,
                },
                "calories": 165,
                "carbohydrates": 0,
                "createdAt": 1000,
                "fat": 3.6,
                "id": "2",
                "name": "Chicken Breast",
                "protein": 31,
                "timesUsed": 1,
                "unit": "grams",
              },
            ],
            "total": Object {
              "calories": 533,
              "carbohydrates": 46.5,
              "fat": 18,
              "protein": 41.2,
            },
            "unit": "lb",
            "weight": 202,
          }
    }
    onKgsSelected={[Function]}
    onLbsSelected={[Function]}
    onRemove={[Function]}
    onSubmit={[Function]}
    onWeightChange={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="currentLog__weight"
>
          <div>
                    <input
                              className="currentLog__input"
                              name="weight"
                              onChange={[Function]}
                              placeholder="Weight"
                              type="text"
                              value={202}
                    />
                    <label
                              className="currentLog__label"
                              htmlFor="weight"
                    >
                              Weight
                    </label>
          </div>
          <div
                    className="currentLog__radio-group"
          >
                    <input
                              checked={false}
                              className="currentLog__radio-input"
                              id="lb"
                              name="unit"
                              onChange={[Function]}
                              type="radio"
                              value="lbs"
                    />
                    <label
                              className="currentLog__radio-label"
                              htmlFor="lb"
                    >
                              <span
                                        className="currentLog__radio-button"
                              />
                              lbs
                    </label>
          </div>
          <div
                    className="currentLog__radio-group"
          >
                    <input
                              checked={false}
                              className="currentLog__radio-input"
                              id="kg"
                              name="unit"
                              onChange={[Function]}
                              type="radio"
                              value="kgs"
                    />
                    <label
                              className="currentLog__radio-label"
                              htmlFor="kg"
                    >
                              <span
                                        className="currentLog__radio-button"
                              />
                              kgs
                    </label>
          </div>
</div>,
        <div
          className="currentLog__btn"
>
          <div>
                    <button
                              className="btn btn--form"
                              onClick={[Function]}
                              type="button"
                    >
                              Remove
                    </button>
          </div>
          <div>
                    <button
                              className="btn btn--form btn--primary"
                              onClick={[Function]}
                    >
                              Submit
                    </button>
          </div>
</div>,
      ],
      "className": "currentLog",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <input
                            className="currentLog__input"
                            name="weight"
                            onChange={[Function]}
                            placeholder="Weight"
                            type="text"
                            value={202}
              />
              <label
                            className="currentLog__label"
                            htmlFor="weight"
              >
                            Weight
              </label>
</div>,
            <div
              className="currentLog__radio-group"
>
              <input
                            checked={false}
                            className="currentLog__radio-input"
                            id="lb"
                            name="unit"
                            onChange={[Function]}
                            type="radio"
                            value="lbs"
              />
              <label
                            className="currentLog__radio-label"
                            htmlFor="lb"
              >
                            <span
                                          className="currentLog__radio-button"
                            />
                            lbs
              </label>
</div>,
            <div
              className="currentLog__radio-group"
>
              <input
                            checked={false}
                            className="currentLog__radio-input"
                            id="kg"
                            name="unit"
                            onChange={[Function]}
                            type="radio"
                            value="kgs"
              />
              <label
                            className="currentLog__radio-label"
                            htmlFor="kg"
              >
                            <span
                                          className="currentLog__radio-button"
                            />
                            kgs
              </label>
</div>,
          ],
          "className": "currentLog__weight",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  className="currentLog__input"
                  name="weight"
                  onChange={[Function]}
                  placeholder="Weight"
                  type="text"
                  value={202}
/>,
                <label
                  className="currentLog__label"
                  htmlFor="weight"
>
                  Weight
</label>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "currentLog__input",
                  "name": "weight",
                  "onChange": [Function],
                  "placeholder": "Weight",
                  "type": "text",
                  "value": 202,
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Weight",
                  "className": "currentLog__label",
                  "htmlFor": "weight",
                },
                "ref": null,
                "rendered": "Weight",
                "type": "label",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  checked={false}
                  className="currentLog__radio-input"
                  id="lb"
                  name="unit"
                  onChange={[Function]}
                  type="radio"
                  value="lbs"
/>,
                <label
                  className="currentLog__radio-label"
                  htmlFor="lb"
>
                  <span
                                    className="currentLog__radio-button"
                  />
                  lbs
</label>,
              ],
              "className": "currentLog__radio-group",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "checked": false,
                  "className": "currentLog__radio-input",
                  "id": "lb",
                  "name": "unit",
                  "onChange": [Function],
                  "type": "radio",
                  "value": "lbs",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span
                      className="currentLog__radio-button"
/>,
                    "lbs",
                  ],
                  "className": "currentLog__radio-label",
                  "htmlFor": "lb",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "currentLog__radio-button",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "span",
                  },
                  "lbs",
                ],
                "type": "label",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  checked={false}
                  className="currentLog__radio-input"
                  id="kg"
                  name="unit"
                  onChange={[Function]}
                  type="radio"
                  value="kgs"
/>,
                <label
                  className="currentLog__radio-label"
                  htmlFor="kg"
>
                  <span
                                    className="currentLog__radio-button"
                  />
                  kgs
</label>,
              ],
              "className": "currentLog__radio-group",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "checked": false,
                  "className": "currentLog__radio-input",
                  "id": "kg",
                  "name": "unit",
                  "onChange": [Function],
                  "type": "radio",
                  "value": "kgs",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span
                      className="currentLog__radio-button"
/>,
                    "kgs",
                  ],
                  "className": "currentLog__radio-label",
                  "htmlFor": "kg",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "currentLog__radio-button",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "span",
                  },
                  "kgs",
                ],
                "type": "label",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <button
                            className="btn btn--form"
                            onClick={[Function]}
                            type="button"
              >
                            Remove
              </button>
</div>,
            <div>
              <button
                            className="btn btn--form btn--primary"
                            onClick={[Function]}
              >
                            Submit
              </button>
</div>,
          ],
          "className": "currentLog__btn",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="btn btn--form"
                onClick={[Function]}
                type="button"
>
                Remove
</button>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Remove",
                "className": "btn btn--form",
                "onClick": [Function],
                "type": "button",
              },
              "ref": null,
              "rendered": "Remove",
              "type": "button",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="btn btn--form btn--primary"
                onClick={[Function]}
>
                Submit
</button>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Submit",
                "className": "btn btn--form btn--primary",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Submit",
              "type": "button",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="currentLog__weight"
>
            <div>
                        <input
                                    className="currentLog__input"
                                    name="weight"
                                    onChange={[Function]}
                                    placeholder="Weight"
                                    type="text"
                                    value={202}
                        />
                        <label
                                    className="currentLog__label"
                                    htmlFor="weight"
                        >
                                    Weight
                        </label>
            </div>
            <div
                        className="currentLog__radio-group"
            >
                        <input
                                    checked={false}
                                    className="currentLog__radio-input"
                                    id="lb"
                                    name="unit"
                                    onChange={[Function]}
                                    type="radio"
                                    value="lbs"
                        />
                        <label
                                    className="currentLog__radio-label"
                                    htmlFor="lb"
                        >
                                    <span
                                                className="currentLog__radio-button"
                                    />
                                    lbs
                        </label>
            </div>
            <div
                        className="currentLog__radio-group"
            >
                        <input
                                    checked={false}
                                    className="currentLog__radio-input"
                                    id="kg"
                                    name="unit"
                                    onChange={[Function]}
                                    type="radio"
                                    value="kgs"
                        />
                        <label
                                    className="currentLog__radio-label"
                                    htmlFor="kg"
                        >
                                    <span
                                                className="currentLog__radio-button"
                                    />
                                    kgs
                        </label>
            </div>
</div>,
          <div
            className="currentLog__btn"
>
            <div>
                        <button
                                    className="btn btn--form"
                                    onClick={[Function]}
                                    type="button"
                        >
                                    Remove
                        </button>
            </div>
            <div>
                        <button
                                    className="btn btn--form btn--primary"
                                    onClick={[Function]}
                        >
                                    Submit
                        </button>
            </div>
</div>,
        ],
        "className": "currentLog",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <input
                                className="currentLog__input"
                                name="weight"
                                onChange={[Function]}
                                placeholder="Weight"
                                type="text"
                                value={202}
                />
                <label
                                className="currentLog__label"
                                htmlFor="weight"
                >
                                Weight
                </label>
</div>,
              <div
                className="currentLog__radio-group"
>
                <input
                                checked={false}
                                className="currentLog__radio-input"
                                id="lb"
                                name="unit"
                                onChange={[Function]}
                                type="radio"
                                value="lbs"
                />
                <label
                                className="currentLog__radio-label"
                                htmlFor="lb"
                >
                                <span
                                                className="currentLog__radio-button"
                                />
                                lbs
                </label>
</div>,
              <div
                className="currentLog__radio-group"
>
                <input
                                checked={false}
                                className="currentLog__radio-input"
                                id="kg"
                                name="unit"
                                onChange={[Function]}
                                type="radio"
                                value="kgs"
                />
                <label
                                className="currentLog__radio-label"
                                htmlFor="kg"
                >
                                <span
                                                className="currentLog__radio-button"
                                />
                                kgs
                </label>
</div>,
            ],
            "className": "currentLog__weight",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    className="currentLog__input"
                    name="weight"
                    onChange={[Function]}
                    placeholder="Weight"
                    type="text"
                    value={202}
/>,
                  <label
                    className="currentLog__label"
                    htmlFor="weight"
>
                    Weight
</label>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "currentLog__input",
                    "name": "weight",
                    "onChange": [Function],
                    "placeholder": "Weight",
                    "type": "text",
                    "value": 202,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Weight",
                    "className": "currentLog__label",
                    "htmlFor": "weight",
                  },
                  "ref": null,
                  "rendered": "Weight",
                  "type": "label",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    checked={false}
                    className="currentLog__radio-input"
                    id="lb"
                    name="unit"
                    onChange={[Function]}
                    type="radio"
                    value="lbs"
/>,
                  <label
                    className="currentLog__radio-label"
                    htmlFor="lb"
>
                    <span
                                        className="currentLog__radio-button"
                    />
                    lbs
</label>,
                ],
                "className": "currentLog__radio-group",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "checked": false,
                    "className": "currentLog__radio-input",
                    "id": "lb",
                    "name": "unit",
                    "onChange": [Function],
                    "type": "radio",
                    "value": "lbs",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <span
                        className="currentLog__radio-button"
/>,
                      "lbs",
                    ],
                    "className": "currentLog__radio-label",
                    "htmlFor": "lb",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "currentLog__radio-button",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "span",
                    },
                    "lbs",
                  ],
                  "type": "label",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    checked={false}
                    className="currentLog__radio-input"
                    id="kg"
                    name="unit"
                    onChange={[Function]}
                    type="radio"
                    value="kgs"
/>,
                  <label
                    className="currentLog__radio-label"
                    htmlFor="kg"
>
                    <span
                                        className="currentLog__radio-button"
                    />
                    kgs
</label>,
                ],
                "className": "currentLog__radio-group",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "checked": false,
                    "className": "currentLog__radio-input",
                    "id": "kg",
                    "name": "unit",
                    "onChange": [Function],
                    "type": "radio",
                    "value": "kgs",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <span
                        className="currentLog__radio-button"
/>,
                      "kgs",
                    ],
                    "className": "currentLog__radio-label",
                    "htmlFor": "kg",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "currentLog__radio-button",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "span",
                    },
                    "kgs",
                  ],
                  "type": "label",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <button
                                className="btn btn--form"
                                onClick={[Function]}
                                type="button"
                >
                                Remove
                </button>
</div>,
              <div>
                <button
                                className="btn btn--form btn--primary"
                                onClick={[Function]}
                >
                                Submit
                </button>
</div>,
            ],
            "className": "currentLog__btn",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  className="btn btn--form"
                  onClick={[Function]}
                  type="button"
>
                  Remove
</button>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Remove",
                  "className": "btn btn--form",
                  "onClick": [Function],
                  "type": "button",
                },
                "ref": null,
                "rendered": "Remove",
                "type": "button",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  className="btn btn--form btn--primary"
                  onClick={[Function]}
>
                  Submit
</button>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Submit",
                  "className": "btn btn--form btn--primary",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Submit",
                "type": "button",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
